app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -427,6 +439,64 @@ public void onClick(View view) {
                 setUpGravatar(message, messageHolder);
                 setUpTime(message, messageHolder);
                 setUpStar(message, messageHolder);
+                try {
+                    messageHolder.reactionsTable.removeAllViews();
```

Comment:
`recylerview` does a perf optimisation where views are reused instead of creating new ones. Hence to avoid any erroneous display of reactions on a message tile, the reactions property is reset from removing all views from the reactions table.